### PR TITLE
fix for #276

### PR DESCRIPTION
**Pull request by 3ll3d00d**

### CODE REVIEW

File: DisplayCAL/display_cal.py [hunk 1]
```diff
@@ -8797,7 +8797,7 @@ def measurement_report_handler(self, event, path=None):
                 mprof = profile
         apply_map = (
             use_sim
-            and mprof.colorSpace == "RGB"
+            and mprof.colorSpace == b"RGB"
```
Comment (3ll3d00d) [inline DisplayCAL/display_cal.py:8800]: this was the cause of the verification report problem

File: DisplayCAL/xh_filebrowsebutton.py [hunk 2]
```diff
@@ -43,6 +43,10 @@ def DoCreateResource(self):
             name=self.GetName(),
         )
         self.SetupWindow(w)
+        if self.GetBool("hidden") and w.Shown:
```
Comment (3ll3d00d) [inline DisplayCAL/xh_filebrowsebutton.py:46]: this was the cause of the extra dropdown which should have been hidden, I do not know wxPython at all and can find v few resources on it & could not work out why SetupWindow was no longer hiding the control as it does on the python2  build so added this workaround here + to all similar custom controls. If anyone knows wxPython and can say how to actually fix the problem, happy to change htis.

File: misc/xrced_plugins/xh_floatspin.py [hunk 3]
```diff
@@ -2,60 +2,65 @@
 
 import wx
 import wx.xrc as xrc
+
```
Comment (3ll3d00d) [inline misc/xrced_plugins/xh_floatspin.py:5]: most changes in here due to black reformat

**Comment by 3ll3d00d** [comment]
the build failures appear to be environmental issues (or at least I can't see how my change would cause the pytest task to segfault)

File: DisplayCAL/xh_filebrowsebutton.py (hunk 2 above)
Comment (p5k369) [reply DisplayCAL/xh_filebrowsebutton.py:46]: I'm not experienced in wxpython at all, however it looks a bit strange, like if an argument is not considered anymore.
>SetupWindow was no longer hiding the control as it does on the python2 

it is very likely that this has more to do with the recent wxpython version we are using than with python3 I think

File: DisplayCAL/xh_filebrowsebutton.py [hunk 4]
```diff
@@ -43,6 +43,10 @@ def DoCreateResource(self):
...
+            from log import safe_print
```
Comment (p5k369) [inline DisplayCAL/xh_filebrowsebutton.py:47]: This is unnecessary, try to group your imports. Conditional imports may make sense to address different architectures or if something like a fallback can be used, but not here.
https://peps.python.org/pep-0008/#imports

File: DisplayCAL/display_cal.py [hunk 5]
```diff
@@ -18147,11 +18147,11 @@ def load_cal_handler(
                     if 'HIRES_B2A "YES"' not in ti3_lines:
                         setcfg("profile.b2a.hires", 0)
                     setcfg("profile.b2a.hires.smooth", 0)
+                simset = False  # Only HDR 3D LUTs will have this set
```
Comment (p5k369) [inline DisplayCAL/display_cal.py:18150]: simset has to be set to False even if "BEGIN_DATA_FORMAT" was not found in ti3. Is that intentional?

Comment (3ll3d00d) [reply DisplayCAL/display_cal.py:18150]: intentional, I noticed it blowing during testing and then I can see my IDE flags it 

image

File: DisplayCAL/xh_filebrowsebutton.py (hunk 4 above)
Comment (3ll3d00d) [reply DisplayCAL/xh_filebrowsebutton.py:47]: it's what I thought also (re wxpython), couldn't find anything in https://wxpython.org/pages/changes/index.html though and there seems relatively little info on wxpython around so workaround seemed fair to me

File: DisplayCAL/xh_filebrowsebutton.py (hunk 2 above)
Comment (3ll3d00d) [reply DisplayCAL/xh_filebrowsebutton.py:46]: it's what I thought also (re wxpython), couldn't find anything in https://wxpython.org/pages/changes/index.html though and there seems relatively little info on wxpython around so workaround seemed fair to me

File: DisplayCAL/xh_filebrowsebutton.py (hunk 4 above)
Comment (3ll3d00d) [reply DisplayCAL/xh_filebrowsebutton.py]: it's fair point, running those locally I see the segfault is in wx

```
[50899.035449] python[431559]: segfault at 140 ip 00007fd49b8a163d sp 00007ffdd5ccdbf0 error 4 in libwx_baseu-3.2.so.0[7fd49b87c000+187000] likely on CPU 8 (core 2, socket 0)
```

running the same on develop doesn't fail

I haven't seen any such failure running the actual app with my changes

on my system (debian trixie), `libwx_baseu-3.2.so.0` appears to be part of

```
libwxbase3.2-1/testing,unstable,now 3.2.2+dfsg-2 amd64 [installed,automatic]
  wxBase library (runtime) - non-GUI support classes of wxWidgets toolkit
```

in my venv I have

`wxPython 4.2.0 gtk3 (phoenix) wxWidgets 3.2.0`

Comment (3ll3d00d) [reply DisplayCAL/xh_filebrowsebutton.py]: I pushed a change to move the import and now it works :shrug:

**Review by p5k369 — APPROVED** [review]

**Pull request merged**